src/NuGet.Clients/NuGet.PackageManagement.VisualStudio/Telemetry/VsProjectBuildPropertiesTelemetry.cs

```diff
@@ -15,49 +18,73 @@ namespace NuGet.PackageManagement.VisualStudio.Telemetry
+        ConcurrentDictionary<string, ConcurrentDictionary<string, ApiUsage>> _projectSystems = new(StringComparer.OrdinalIgnoreCase);
 
         public VsProjectBuildPropertiesTelemetry()
         {
             InstanceCloseTelemetryEmitter.AddEventsOnShutdown += AddEventsOnShutdown;
         }
 
-        public void OnDteUsed(string[] projectTypeGuids)
+        public void OnDteUsed(string propertyName, string[] projectTypeGuids)
         {
             foreach (var typeGuid in projectTypeGuids)
             {
-                var apiUsage = _apiUsages.GetOrAdd(typeGuid, NewApiUsage);
+                ApiUsage apiUsage = GetApiUsage(propertyName, typeGuid);
                 apiUsage.DteUsed = true;
             }
         }
 
-        public void OnPropertyStorageUsed(string[] projectTypeGuids)
+        public void OnPropertyStorageUsed(string propertyName, string[] projectTypeGuids)
         {
             foreach (var typeGuid in projectTypeGuids)
             {
-                var apiUsage = _apiUsages.GetOrAdd(typeGuid, NewApiUsage);
+                ApiUsage apiUsage = GetApiUsage(propertyName, typeGuid);
                 apiUsage.PropertyStorageUsed = true;
             }
         }
 
-        private static ApiUsage NewApiUsage(string projectTypeGuid)
+        private ApiUsage GetApiUsage(string propertyName, string projectTypeGuid)
+        {
+            ConcurrentDictionary<string, ApiUsage> propertyNames = _projectSystems.GetOrAdd(projectTypeGuid, CreatePropertyNamesDictionary);
+            ApiUsage apiUsage = propertyNames.GetOrAdd(propertyName, CreateApiUsage, projectTypeGuid);
+            return apiUsage;
+        }
+
+        private static ConcurrentDictionary<string, ApiUsage> CreatePropertyNamesDictionary(string _)
+        {
+            return new ConcurrentDictionary<string, ApiUsage>();
+        }
+
+        private static ApiUsage CreateApiUsage(string propertyName, string projectTypeGuid)
         {
-            var guid = projectTypeGuid.ToUpperInvariant();
             return new ApiUsage()
             {
-                ProjectType = guid
+                ProjectType = projectTypeGuid,
+                PropertyName = propertyName
             };
         }
 
         internal void AddEventsOnShutdown(object sender, TelemetryEvent e)
         {
-            var entries = _apiUsages.Values.ToList();
-            e.ComplexData["ProjectBuildProperties"] = entries;
+            int count = _projectSystems.Values.Sum(p => p.Count);
+            List<ApiUsage> items = new(count);
+
+            foreach (var projectSystem in _projectSystems)
+            {
+                foreach (var property in projectSystem.Value)
+                {
+                    ApiUsage apiUsage = property.Value;
+                    items.Add(property.Value);
+                }
+            }
+
+            e.ComplexData["ProjectBuildProperties"] = items;
```

Comment:
I'm not sure if I should change the property name, so our queries can easily filter out old telemetry without the property name. Or if we should just filter out telemetry where the property name is missing.

Reply:
I'm not too concerned with the breaking change. We can handle it on the telemetry end, but also we're very likely to make decisions based off of the latest usage of the tooling anyways, since that's the one with more actionable telemetry.